### DISCUSSION TITLE
CNF configuration

### DISCUSSION THREAD

**Discussion by murali-reddy** [lfn-cnti/bestpractices#40] [General]
starting a new discussion to get perspective on how CNF vendors are approaching the problem of configuration changes? What can we consider cloud native way of making configuration changes to CNF's? In ideal scenario can CNF's be treated as immutable? and any configuration change would result in spinning a new CNF with the desired configuration change and old CNF is destroyed after draining the traffic to them. Alternativley CNF's can be updated in-place with netconf etc.

**Comment by xmulligan** [comment]
[Ian] This will overlap with the packaging discussion

**Reply by rmerz** [reply]
Isn't this related to how session management is performed?
Taking a 5G core UPF as example: if the subscriber session is gracefully handled by whatever logic is available before tearing down the CNF, I think treating the CNF as immutable is feasible.

**Comment by vukg** [comment]
It is also related to requirement that CNF has to handle failure of single or multiple cluster nodes without interruptions in service (node disruption budget and related pod disruption budgets). If CNF satisfies this it can also satisfy immutable configs. I have user story for that.
